### PR TITLE
Add tests for the conversions crate

### LINKED ISSUE #599: Add tests for the `conversions` crate
- `assert x.to_y().to_x() == x`
- create new objects from `1`

### PR DESCRIPTION
**Pull request by devnet0x**
Closes #599 

## Introduced changes

Add tests for the conversions crate

- class_hash
- contract_address
- felt252
- field_element
- nonce
- short_string
- stark_felt
- stark_hash

## Breaking changes

N/A

## Checklist

- [X] Linked relevant issue
- [ N/A] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [N/A] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by Arcticae — CHANGES_REQUESTED** [review]
This seems fine by me, although please add more values from which we test the conversions (the ones that could be problematic, boundary values etc.)

**Review by war-in — COMMENTED** [review]
Do you think some tests for errors would be suitable?

**Comment by devnet0x** [comment]
Added boundary values and error tests. Please your review 🔎

File: crates/conversions/tests/e2e/short_string.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        );
+        assert_eq!(short_string, short_string.to_nonce().to_short_string());
+        assert_eq!(short_string, short_string.to_stark_felt().to_short_string());
+        assert_eq!(short_string, short_string.to_stark_hash().to_short_string());
+    }
+
+    #[test]
+    fn test_short_string_conversions_limit() {
+        // 32 characters.
+        let short_string = "1234567890123456789012345678912".to_string();
```
Comment (war-in) [inline crates/conversions/tests/e2e/short_string.rs:46]: I think it's 31 characters ;)

File: crates/conversions/tests/helpers/hex.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+pub fn str_hex_to_stark_felt(hex_string: &str) -> StarkFelt {
+    let hex_string = hex_string.strip_prefix("0x").unwrap().to_string();
+    let mut byte_array = [0u8; 32];
+    for i in 0..32 {
+        let byte_str = &hex_string[i * 2..i * 2 + 2];
+        byte_array[i] = u8::from_str_radix(byte_str, 16).unwrap();
+    }
+
+    StarkFelt::new(byte_array).unwrap()
+}
```
Comment (war-in) [inline crates/conversions/tests/helpers/hex.rs:15]: All those 3 functions are almost the same. It would be good to extract duplicated code

File: crates/conversions/tests/e2e/class_hash.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        assert_eq!(class_hash, class_hash.to_stark_hash().to_class_hash());
+
+        // PATRICIA_KEY_UPPER_BOUND for contract_address from starknet_api-0.4.1/src/core.rs:156
+        max_value = "0x07ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+        class_hash = ClassHash(str_hex_to_stark_felt(max_value));
+        assert_eq!(class_hash, class_hash.to_contract_address().to_class_hash());
+
+        // Unknown source for this value, founded by try and error(cairo-lang-runner-2.2.0/src/short_string.rs).
+        max_value = "0x0777777777777777777777777777777777777f7f7f7f7f7f7f7f7f7f7f7f7f7f";
+        class_hash = ClassHash(str_hex_to_stark_felt(max_value));
```
Comment (war-in) [inline crates/conversions/tests/e2e/class_hash.rs:55]: Is it the max value for a short string?

File: crates/conversions/tests/e2e/short_string.rs (hunk 1 above)
Comment (devnet0x) [reply crates/conversions/tests/e2e/short_string.rs:46]: fixed

File: crates/conversions/tests/e2e/class_hash.rs (hunk 3 above)
Comment (devnet0x) [reply crates/conversions/tests/e2e/class_hash.rs:55]: yes, i don't know why just founded by try and error. I asked in cairo stars but don't have a response why this is the max value?

File: crates/conversions/tests/helpers/hex.rs (hunk 2 above)
Comment (devnet0x) [reply crates/conversions/tests/helpers/hex.rs]: fixed

File: crates/conversions/tests/e2e/short_string.rs [hunk 4]
```diff
@@ -43,7 +43,7 @@ mod tests_short_string {
     #[test]
     fn test_short_string_conversions_limit() {
         // 32 characters.
```
Comment (war-in) [inline crates/conversions/tests/e2e/short_string.rs:45]:
```suggestion
        // 31 characters.
```

File: crates/conversions/tests/e2e/short_string.rs [hunk 5]
```diff
@@ -43,7 +43,7 @@ mod tests_short_string {
...
-        let short_string = "1234567890123456789012345678912".to_string();
+        let short_string = "123456789012345678901234567891".to_string();
```
Comment (war-in) [inline crates/conversions/tests/e2e/short_string.rs:46]:
```suggestion
        let short_string = "1234567890123456789012345678901".to_string();
```

**Review by war-in — COMMENTED** [review]
I meant that 😅

**Review by war-in — APPROVED** [review]
Good job 🔥 Let's wait for the @Arcticae and merge it :rocket:

**Review by Arcticae — APPROVED** [review]
